src/processor/operator/persistent/reader/csv/driver.cpp

```diff
@@ -239,5 +240,64 @@ bool SniffCSVNameAndTypeDriver::addValue(uint64_t rowNum, common::column_id_t co
     return true;
 }
 
+SniffCSVHeaderDriver::SniffCSVHeaderDriver(SerialCSVReader* reader,
+    const function::ScanTableFuncBindInput* /*bindInput*/,
+    const std::vector<std::pair<std::string, common::LogicalType>>& typeDetected)
+    : SerialParsingDriver(getDummyDataChunk(), reader, DriverType::SNIFF_CSV_HEADER) {
+    for (auto i = 0u; i < typeDetected.size(); i++) {
+        columns.push_back({typeDetected[i].first, typeDetected[i].second.copy()});
+    }
+}
+
+bool SniffCSVHeaderDriver::done(uint64_t rowNum) const {
+    // Only read the firt line.
+    return (0 < rowNum);
+}
+
+bool SniffCSVHeaderDriver::addValue(uint64_t /*rowNum*/, common::column_id_t columnIdx,
+    std::string_view value) {
+    uint64_t length = value.length();
+    if (length == 0 && columnIdx == 0) {
+        rowEmpty = true;
+    } else {
+        rowEmpty = false;
+    }
+    if (columnIdx == reader->getNumColumns() && length == 0) {
+        // skip a single trailing delimiter in last columnIdx
+        return true;
+    }
+
+    // reading the header
+    LogicalType columnType(LogicalTypeID::ANY);
+
+    try {
+        columnType = function::inferMinimalTypeFromString(value);
```

Comment:
I suggest we add another function inferMinimalTypeFromString which returns true if the inferMinimalTypeFromString succeeds otherwise returns false.
The cost of try/catch statement is quite high.

Reply:
I recheck the inferMinimalTypeFromString function and found we don't really need the try and catch, itself is error free. I removed this